src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const cloneActions = [];
+  let currentState = Object.assign({}, state);
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        currentState = Object.assign({}, currentState, action.extraData);
+
+        break;
+
+      case 'removeProperties':
+        currentState = Object.assign({}, currentState);
+
+        for (const key of action.keysToRemove) {
+          delete currentState[key];
+        }
+
+        break;
+
+      case 'clear':
+        currentState = {};
+
+        break;
+
+      default:
+        break;
+    }
+
+    cloneActions.push(Object.assign({}, currentState));
```

Comment:
You have repeatedly cloned the 'currentState' object throughout your code and in the beginning of every action. This could potentially use a lot of memory if the state and actions arrays are large. A more memory-efficient approach might be to only clone the state when you are about to modify it.

Reply:
Did I understand the wish correctly?